Tests/MongoSwiftTests/AsyncAwaitTests.swift

```diff
@@ -114,4 +115,168 @@ final class AsyncAwaitTests: MongoSwiftTestCase {
+                }
+
+                // Cancel the child task, which should break us out of the loop.
+                cursorTask.cancel()
+
+                // If the test got to the end, it means we successfully broke out of the loop and left the group.
+                group.wait()
+            }
+        }
+    }
+}
+
+final class ChangeStreamAsyncAwaitTests: MongoSwiftTestCase {
+    func testIteration() throws {
+        testAsync {
+            try await self.withTestNamespace { client, _, coll in
+                guard try await client.supportsChangeStreamOnCollection() else {
+                    printSkipMessage(testName: self.name, reason: "Requires change streams support")
+                    return
+                }
+
+                let changeStream = try await coll.watch()
+
+                let docsToInsert: [BSONDocument] = [["_id": 1], ["_id": 2], ["_id": 3]]
+                for doc in docsToInsert {
+                    try await coll.insertOne(doc)
+                }
+
+                // manual iteration with next()
+                let first = try await changeStream.next()
+                expect(first?.fullDocument).to(equal(docsToInsert[0]))
+                let second = try await changeStream.next()
+                expect(second?.fullDocument).to(equal(docsToInsert[1]))
+                let third = try await changeStream.next()
+                expect(third?.fullDocument).to(equal(docsToInsert[2]))
+                let isAlive = try await changeStream.isAlive()
+                expect(isAlive).to(beTrue())
+            }
+        }
+    }
+
+    // Test that a change stream that is continually polling the server can be killed by cancelling the parent Task.
+    func testHandlesTaskCancellation() throws {
+        testAsync {
+            try await self.withTestNamespace { client, _, coll in
+                guard try await client.supportsChangeStreamOnCollection() else {
+                    printSkipMessage(testName: self.name, reason: "Requires change streams support")
+                    return
+                }
+
+                let group = DispatchGroup()
+                group.enter()
+
+                let eventCount = NIOAtomic<Int>.makeAtomic(value: 0)
+                let calledWatch = NIOAtomic<Bool>.makeAtomic(value: false)
+
+                let csTask = Task.detached {
+                    do {
+                        let changeStream = try await coll.watch()
+                        calledWatch.store(true)
+                        for try await _ in changeStream {
+                            _ = eventCount.add(1)
+                        }
+                    } catch {
+                        XCTFail("\(error)")
+                    }
+                    group.leave()
+                }
+
+                // Wait until the change stream has been opened before we start inserting data.
+                try await assertIsEventuallyTrue(description: "change stream should be started") {
```

Comment:
this was a race condition that for whatever reason only showed up on server versions up to but not including 4.4, where occasionally the docs would get inserted below before we managed to create the change stream and so no events would be observed.